### PR TITLE
Python: rework OutputId so it works in deserialization

### LINKED ISSUE #2020: Fix Python OutputId deserialization
Change Python OutputId deserialization so it works for OutputMetadata, OutputWithMetadata, currently fails with `AttributeError: 'str' object has no attribute 'keys'`

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Python: rework OutputId so it works in deserialization

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/2020

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Running tests

### CODE REVIEW

**Review by DaughterOfMars — DISMISSED** [review]
This is fine I guess. What was the reason it wouldn't work as a class with two fields?

**Comment by Thoralf-M** [comment]
> This is fine I guess. What was the reason it wouldn't work as a class with two fields?

Because it needs to be a dict for multiple fields and then it fails to deserialize because the field in the data is just a string `AttributeError: 'str' object has no attribute 'keys'`
This could probably also solved in some other way with a custom deserialization method, but didn't found how this can be done and to me this now also seems to be cleaner overall

**Review by Alex6323 — DISMISSED** [review]

File: bindings/python/iota_sdk/types/output_id.py [hunk 1]
```diff
@@ -2,64 +2,66 @@
...
-        # Validate that it has only valid hex characters
-        int(transaction_id[2:], 16)
-        output_index_hex = (output_index).to_bytes(2, "little").hex()
-        self.output_id = transaction_id + output_index_hex
-        self.transaction_id = transaction_id
-        self.output_index = output_index
+                'output_id length must be 78 characters with 0x prefix')
+        if not output_id.startswith('0x'):
+            raise ValueError('output_id must start with 0x')
+        instance = super().__new__(cls, output_id)
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/output_id.py:27]: What is `super` here? `str`?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/output_id.py:27]: Yes

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/output_id.py:27]: So apart from length and 0x, we don't really check that it's hexadecimal ?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/output_id.py:27]: No, added

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**